src/main/cpp/RLBotInterface/RLBotInterface/GameFunctions/GameFunctions.cpp

```diff
@@ -8,218 +8,19 @@
-												}
+#include "..\CallbackProcessor\SharedMemoryDefinitions.hpp"
 
 namespace GameFunctions
 {
-	bool isValidName(wchar_t* pName)
-	{
-		for (int i = 0; i < sizeof(PlayerConfiguration::Name) / sizeof(wchar_t); i++)
-		{
-			wchar_t wideChar = pName[i];
-
-			if (wideChar == L'\0')
-				return true;
-
-			if (wideChar < L' ' || wideChar > L'~')
-				return false;
-		}
-
-		return false;
-	}
-
-	RLBotCoreStatus checkPlayerConfiguration(PlayerConfiguration playerConfiguration[CONST_MaxPlayers], int numPlayers)
-	{
-		if (numPlayers < 0 || numPlayers > CONST_MaxPlayers)
-			return RLBotCoreStatus::InvalidNumPlayers;
-
-		for (int i = 0; i < numPlayers; i++)
-		{
-			PlayerConfiguration& playerConfig = playerConfiguration[i];
-
-			if (playerConfig.Bot)
-			{
-				if (!playerConfig.RLBotControlled)
-				{
-					if (playerConfig.BotSkill < 0.0f || playerConfig.BotSkill > 1.0f)
-						return RLBotCoreStatus::InvalidBotSkill;
-				}
-
-				if (!isValidName(playerConfig.Name))
-					RLBotCoreStatus::InvalidName;
-			}
-
-			if (playerConfig.Team < 0 || playerConfig.Team > 1)
-				return RLBotCoreStatus::InvalidTeam;
-
-			if (playerConfig.TeamColorID < 0 || playerConfig.TeamColorID > 27)
-				return RLBotCoreStatus::InvalidTeamColorID;
-
-			if (playerConfig.CustomColorID < 0 || playerConfig.CustomColorID > 104)
-				return RLBotCoreStatus::InvalidTeamColorID;
-
-			//Todo: Add items validation
-		}
-
-		return RLBotCoreStatus::Success;
-	}
-
-	RLBotCoreStatus checkInputConfiguration(const PlayerInput& playerInput)
-	{
-		if (playerInput.Throttle < -1.0f || playerInput.Throttle > 1.0f)
-			return RLBotCoreStatus::InvalidThrottle;
-
-		if (playerInput.Steer < -1.0f || playerInput.Steer > 1.0f)
-			return RLBotCoreStatus::InvalidSteer;
-
-		if (playerInput.Pitch < -1.0f || playerInput.Pitch > 1.0f)
-			return RLBotCoreStatus::InvalidPitch;
-
-		if (playerInput.Yaw < -1.0f || playerInput.Yaw > 1.0f)
-			return RLBotCoreStatus::InvalidYaw;
-
-		if (playerInput.Roll < -1.0f || playerInput.Roll > 1.0f)
-			return RLBotCoreStatus::InvalidRoll;
-
-		return RLBotCoreStatus::Success;
-	}
-
-	RLBotCoreStatus checkQuickChatPreset(QuickChatPreset quickChatPreset)
-	{
-		if (quickChatPreset < 0 || quickChatPreset > QuickChatPreset::MaxQuickChatPresets)
-			return RLBotCoreStatus::InvalidQuickChatPreset;
-
-		return RLBotCoreStatus::Success;
-	}
-
-	
-
-	ByteBuffer fetchByteBufferFromSharedMem(boost::interprocess::shared_memory_object* shm, boost::interprocess::named_sharable_mutex* mtx)
-	{
-		// The lock will be released when this object goes out of scope
-		boost::interprocess::sharable_lock<boost::interprocess::named_sharable_mutex> myLock(*mtx);
-
-		boost::interprocess::offset_t size;
-		shm->get_size(size);
-		if (size == 0)
-		{
-			// Bail out early because mapped_region will freak out if size is zero.
-			ByteBuffer empty;
-			empty.ptr = new char[1]; // Arbitrary valid pointer to an array. We'll be calling delete[] on this later.
```

Comment:
Is this allocation really necessary?

Reply:
This is my code, and I did this because otherwise `delete []` throws an error. There might be a better way to do it.

Reply:
Where is that delete call you are mentioning?

Reply:
There are a couple examples of it visible in the diff. Check here if it's easier: https://github.com/RLBot/RLBot/blob/v4/src/main/cpp/RLBotInterface/RLBotInterface/GameFunctions/GameFunctions.cpp#L204

Reply:
Well I'd rather set it to a nullptr and only call delete when it's not a nullptr. However if it's easier we can leave it as it is for now.

Reply:
Ah, I didn't think of that. I can fix that later.